Activities/Database/ConnectionDialog/UiPath.Data.ConnectionUI.Dialog/Workaround/DbWorkarounds.cs

```diff
@@ -29,13 +29,39 @@ public static void SNILoadWorkaround(bool isWindows = true)
             if(!isWindows)
                 return;
 
-            IntPtr Handle = LoadLibrary(Path.GetFullPath((typeof(DbWorkarounds).Assembly.CodeBase.Replace("file:///", "")) + RelativePath));
+            var asmLocation = GetAssemblyLocation();
+            var path = Path.GetFullPath(asmLocation + RelativePath);
 
+            IntPtr Handle = LoadLibrary(path);
             if (Handle == IntPtr.Zero)
             {
                 int errorCode = Marshal.GetLastWin32Error();
-                throw new Exception(string.Format("Failed to load library (ErrorCode: {0})", errorCode));
+                string errorMessage = string.Format("Failed to load library {0} (ErrorCode: {1})", path, errorCode);
+                throw new Exception(errorMessage);
             }
         }
+
+#if NETCOREAPP
+        private static string GetAssemblyLocation()
+        {
+            return typeof(DbWorkarounds).Assembly.Location;
+        }
+#else
+        private static string GetAssemblyLocation()
+        {
+            //on legacy still use the Assembly.CodeBase
+            //on legacy Assembly.Location returns the shadow copy location and not the original location (so we won't find the sni dll relative to this path)
+            var codeBase = typeof(DbWorkarounds).Assembly.CodeBase;
+            bool isUNCPath = codeBase.StartsWith("file:////");
```

Comment:
should be 4 / or just 3?

Reply:
both
If the Path is UNC type it should have 4, otherwise 3
See these examples
NonUNC(DOS Standard type): file:///C:/Users/viorel.groza/.nuget/packages/uipath.database.activities/1.10.24246-dev.775/lib/net6.0-windows7.0/UiPath.Data.ConnectionUI.Dialog.dll
UNC: file:////dsdprpafile1/RPA_NAS/Dev_Packages/mmenotti/uipath.database.activities/1.10.24242-dev.851/lib/net6.0-windows7.0/UiPath.Database.dll

The problem for UNC paths is that when removing prefix file:/// the remaining path was in the form /path .. which means relative path .. and when we were retrieving the absolute path it would be in the form c:/path (this is the problem that happened in clien't  environment)

Logs from customer:

Here are the logs:
Connect to database: Failed to load library (ErrorCode: 126)
 codeBase file:////dsdprpafile1/RPA_NAS/Dev_Packages/mmenotti/uipath.database.activities/1.10.24242-dev.851/lib/net6.0-windows7.0/UiPath.Database.dll
 escapedCodeBase file:////dsdprpafile1/RPA_NAS/Dev_Packages/mmenotti/uipath.database.activities/1.10.24242-dev.851/lib/net6.0-windows7.0/UiPath.Database.dll
 location \\dsdprpafile1\RPA_NAS\Dev_Packages\mmenotti\uipath.database.activities\1.10.24242-dev.851\lib/net6.0-windows7.0/UiPath.Database.dll
 path D:\dsdprpafile1\RPA_NAS\Dev_Packages\mmenotti\uipath.database.activities\1.10.24242-dev.851\lib\net6.0-windows7.0\runtimes\win-x64\native\Microsoft.Data.SqlClient.SNI.dll
The path for Microsoft.Data.SqlClient.SNI.dll does not exist on D Drive. D drive is where the project is stored (D:\FR_RPA\Tests_MME\Test_DB_Windows\project.json). It looks like it concatenates the Drive lettre of the project and the path where is stored the package \\dsdprpafile1\RPA_NAS\Dev_Packages\mmenotti\uipath.database.activities\1.10.24242-dev.851\l
This is the real path of this file: \\dsdprpafile1\RPA_NAS\Dev_Packages\mmenotti\uipath.database.activities\1.10.24242-dev.851\lib\net6.0-windows7.0\runtimes\win-x64\native\Microsoft.Data.SqlClient.SNI.dll